pkg/util/restoptions/configgetter.go

```diff
@@ -115,30 +112,31 @@ func (g *configRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource)
 	if !specified || configuredCacheSize < 0 {
 		configuredCacheSize = g.defaultCacheSize
 	}
+	storageWithCacher := registry.StorageWithCacher(configuredCacheSize)
 
 	decorator := func(
 		copier runtime.ObjectCopier,
 		storageConfig *storagebackend.Config,
-		requestedSize int,
+		requestedSize *int,
 		objectType runtime.Object,
 		resourcePrefix string,
 		keyFunc func(obj runtime.Object) (string, error),
 		newListFn func() runtime.Object,
 		getAttrsFunc storage.AttrFunc,
 		triggerFn storage.TriggerPublisherFunc,
 	) (storage.Interface, factory.DestroyFunc) {
-		capacity := requestedSize
-		if capacity == UseConfiguredCacheSize {
-			capacity = configuredCacheSize
+		// use the origin default cache size, not the one in registry.StorageWithCacher
```

Comment:
I prefer the `capacity` variable, since after line 130 here, it's no longer `requestedSize`.  It's "the right size".